### PR TITLE
libc/blockstream: add block out stream backend

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Impact
No, new stream implementation

### CODE REVIEW

File: libs/libc/stream/lib_blkoutstream.c [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+ *                lib_blkoutstream_s to be initialized.
+ *
+ * Returned Value:
+ *   None (User allocated instance initialized).
+ *
+ ****************************************************************************/
+
+void lib_blkoutstream_close(FAR struct lib_blkoutstream_s *stream)
+{
+  if (stream)
```
Comment (pkarashchenko) [inline libs/libc/stream/lib_blkoutstream.c:151]: Optional
```suggestion
  if (stream != NULL)
```

File: libs/libc/stream/lib_blkoutstream.c [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    {
+      if (stream->inode)
```
Comment (pkarashchenko) [inline libs/libc/stream/lib_blkoutstream.c:153]: Optional
```suggestion
      if (stream->inode != NULL)
```

File: libs/libc/stream/lib_blkoutstream.c [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+        {
+          close_blockdriver(stream->inode);
+          stream->inode = NULL;
+        }
+
+      if (stream->cache)
```
Comment (pkarashchenko) [inline libs/libc/stream/lib_blkoutstream.c:159]: Optional
```suggestion
      if (stream->cache != NULL)
```

**Review by pkarashchenko — APPROVED** [review]

File: libs/libc/stream/lib_blkoutstream.c (hunk 3 above)
Comment (xiaoxiang781216) [reply libs/libc/stream/lib_blkoutstream.c]: Done.

File: libs/libc/stream/lib_blkoutstream.c (hunk 2 above)
Comment (xiaoxiang781216) [reply libs/libc/stream/lib_blkoutstream.c]: Done.

File: libs/libc/stream/lib_blkoutstream.c (hunk 1 above)
Comment (xiaoxiang781216) [reply libs/libc/stream/lib_blkoutstream.c]: Done.

**Comment by xiaoxiang781216** [comment]
How about we merge this patch?

**Pull request merged**